v0.28 hebrew tokenizer (#1728)
meilisearch/documentation: learn/what_is_meilisearch/language.md

# Language

Meilisearch is multilingual, featuring optimized support for:

- **Any language that uses whitespace to separate words**

> - **Chinese**

Reviewer: I think we should "un-bold" lines 5-8

Author: I went through the same thought process, unbolded them, thought it looked less good, then put the bold back. Anyway I agree that it doesn't match our style guide so consider this change made ✅

- Japanese
- Hebrew

> We aim to provide global language support, and your feedback helps us move closer to that goal. If you notice inconsistencies in your search results or the way your documents are processed, please [open an issue in our tokenizer repo](https://github.com/meilisearch/charabia/issues/new).

Reviewer 2: It's not so easy for the users to know if the tokenizer is the source of the issue, I'd prefer that users create an issue on Meilisearch in order to be moved or copied in the good repository by a core-team member. 😊

```suggestion
We aim to provide global language support, and your feedback helps us move closer to that goal. If you notice inconsistencies in your search results or the way your documents are processed, please [open an issue in Meilisearch repo](https://github.com/meilisearch/meilisearch/issues/new/choose).
```

Reviewer 2: Basically, it's better to redirect users to [Meilisearch issues creation page](https://github.com/meilisearch/meilisearch/issues/new/choose) when they want to create an issue.

The only exception I see is the Language support  where we redirect users:
https://github.com/meilisearch/product/discussions?discussions_q=label%3Aproduct%3Acore%3Atokenizer

Maybe it's a good idea to add a specific redirection in the Meilisearch issues creation page, what do you think @curquiza?

Reviewer 3: Hello! I agree with many: any issue when using Meilisearch should be in the Meilisearch repo. The users cannot really guess if it's related to the tokenizer or not, and it's not their job.
The exception is indeed for the language support request we expect on the product repo -> we've just merged [a PR](https://github.com/meilisearch/meilisearch/pull/2592) to update our issue templates

So, maybe what you want to do (docs team) is only to encourage people to open a language request on the product side.

[Read more about our tokenizer](/learn/advanced/tokenization.md)

## Improving our language support

While we have employees from all over the world at Meilisearch, we don't speak every language. We rely almost entirely on feedback from external contributors to understand how our engine is performing across different languages.

> If you'd like to request optimized support for a language that we don't currently support, please upvote the related [discussion in our product repository](https://github.com/meilisearch/product/discussions?discussions_q=label%3Aproduct%3Acore%3Atokenizer) or [open a new one](https://github.com/meilisearch/product/discussions/new) if it doesn't exist.

Reviewer 2: There is a small mistake in my last proposal, here we create an issue in the good category.

```suggestion
If you'd like to request optimized support for a language that we don't currently support, please upvote the related [discussion in our product repository](https://github.com/meilisearch/product/discussions?discussions_q=label%3Aproduct%3Acore%3Atokenizer) or [open a new one](https://github.com/meilisearch/product/discussions/new?category=feedback-feature-proposal) if it doesn't exist.
```

If you'd like to help by developing a tokenizer pipeline yourself: first of all, thank you! We recommend that you take a look at the [tokenizer contribution guide](https://github.com/meilisearch/charabia/blob/main/CONTRIBUTING.md) before making a PR.

## FAQ

### What do you mean when you say Meilisearch offers _optimized_ support for a language?

Under the hood, Meilisearch relies on tokenizers that identify the most important parts of each document in a given dataset. We currently use four tokenization pipelines:

> - A default one designed for languages that separate words with spaces

Reviewer 4: Pet peeve: I really dislike most occurrences of this pattern: "A/The [adjective] one". I find it so clunky and vague.

Ergo, I'm making a super important and 100% objective suggestion:

```suggestion
- A default pipeline designed for languages that separate words with spaces
```

- A pipeline specifically tailored for Chinese
- A pipeline specifically tailored for Japanese
- A pipeline specifically tailored for Hebrew

### My language does not use whitespace to separate words. Can I still use Meilisearch?

Yes, but search results might be less relevant than in one of the fully optimized languages.

### My language does not use the Roman alphabet. Can I still use Meilisearch?

Yes—our users work with many different alphabets and writing systems, such as Cyrillic, Thai, and Japanese.

### Does Meilisearch plan to support additional languages in the future?

> Yes, we definitely do. The more feedback we get from native speakers, the easier it is for us to understand how to improve performance for those languages. Similarly, the more requests we get to improve support for a specific language, the more likely we are to devote resources to that project.

Reviewer 2: Could we add some links in the same mood as my previous proposal?
